### PR TITLE
metrics diff/show: rollback to rounding fractional digits with precision

### PR DESCRIPTION
**Pull request by skshetry**
* [x] ❗ I have followed the [Contributing to DVC](https://dvc.org/doc/user-guide/contributing/core) checklist.

* [x] 📖 If this PR requires [documentation](https://dvc.org/doc) updates, I have created a separate PR (or issue, at least) in [dvc.org](https://github.com/iterative/dvc.org) and linked it here.

Thank you for the contribution - we'll try to review it as soon as possible. 🙏

### CODE REVIEW

File: dvc/compare.py
Comment (skshetry) [inline dvc/compare.py:184]: I didn't go for  a `f` specifier for rounding because a) we used to use `round` before and, b) `f` is a fixed width formatting, so there could be trailing zeros which `round` does not have.

**Review by efiop — APPROVED** [review]
Thank you!

**Comment by dberenbaum** [comment]
What's the background on the rollback? What's the precision behavior for `exp show` now? I thought this all came up to try to resolve a discrepancy between `metrics diff` and `exp show`?

**Comment by skshetry** [comment]
> What's the background on the rollback? 

@efiop and I had some doubts about the new behaviour, so we decided to rollback to not block the release.

@dberenbaum, I'll leave this up to you now to clarify what should be the expected and correct behaviour for the `--precision` here (I know we have discussed this but would be good to have more closer look?).

Another thing that is unclear to me is, why do we have `--precision` flag? For what purpose do the users use that flag?
Or, is it us, trying to provide a way to change the default behaviour that we apply?

> What's the precision behavior for `exp show` now? 

It is not affected. It's same as before, with it affecting significant digits.

**Comment by dberenbaum** [comment]
> @dberenbaum, I'll leave this up to you now to clarify what should be the expected and correct behaviour for the `--precision` here (I know we have discussed this but would be good to have more closer look?).

Looking at some examples and in Studio, it seems like `g` format is probably best here. Are there specific concerns that were raise about that format or changing formats in general?



> Another thing that is unclear to me is, why do we have `--precision` flag? For what purpose do the users use that flag?
> Or, is it us, trying to provide a way to change the default behaviour that we apply?

I would think it's primary use is to give users some flexibility to make outputs more readable without dvc having to always guess what format is best. One user suggested it could also be useful to filter metrics to only those with a difference greater than the specified precision (see https://github.com/iterative/dvc/discussions/5790), but I lean towards not adopting this interpretation for now.

**Pull request merged**
